### PR TITLE
2019-07-12

### PR DESCRIPTION
**Pull request by Mengqi29**
feat(assignment): Submit assignments of mission 2019-07-10

This is my submission of mission 2019-07-10. It includes:

1. Leetcode 349
2. 349 efficient solution.

### CODE REVIEW

**Comment by Jozdortraz** [comment]
Please merge all commits to a single one @Mengqi29 

@miniwangdali Please help her to follow the pr rule.

File: Assignments/2019-07-10/LC349_Efficient_Solution.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                Nums3.add(nums3[i]);
+            i++;
+        }
+
+        for (int j = 0; j < nums4.length; j++) {
+            Nums4.add(nums4[j]);
+        }
+
+        for (int j = 0; j < Nums3.size(); j++) {
+            if (Nums4.contains(Nums3.get(j))) {
```
Comment (miniwangdali) [inline Assignments/2019-07-10/LC349_Efficient_Solution.java]: hmm，ArrayList的contains效率是O(n)，这样你的总体效率就是O(n * m)了，这样并不高效。

Comment (Mengqi29) [reply Assignments/2019-07-10/LC349_Efficient_Solution.java]: 那感觉我这两种方法效率都应该是O(n*m)，如果是排序排好的话，我之前想了很久没想出来，我再想想！

**Comment by Jozdortraz** [comment]
@Mengqi29  Please resolve conflicts before creating a new pr

File: Assignments/2019-07-10/LC349_Efficient_Solution.java (hunk 1 above)
Comment (miniwangdali) [reply Assignments/2019-07-10/LC349_Efficient_Solution.java]: 没事，想不出来就先等等，等钱老师写了答案了，看了就明白了。
正好，你想想如果有个什么样的数据结构才能提高效率，想一些它的特性。

**Comment by miniwangdali** [comment]
To squash commits:
1. `git rebase -i HEAD~N`, in which `N` is the count of your commits. for this pr, it can be `git rebase -i HEAD~12`.
2. It will give you a few lines with `pick xxxxxxx xxxxxxxxxxxxxxxxxx`. For example, you will see:
```
pick 7b8d653 2019-07-08
pick aefd8fc Create .DS_Store
pick 6bb4440 2019-07-08
pick 41e11ce Assignments
pick 43e4eff 2019-07-08
pick aa63389 2019-07-09
pick a2d2a0c Delete 0-96ppt.pdf
pick c432e56 2019-07-09
pick 2baac34 Delete 0-96ppt.pdf
pick 4617c56 Update .DS_Store
pick cb1ca78 2019-07-10
pick 1291efe 2019-07-11
```
Change them to
```
pick 7b8d653 2019-07-08
f aefd8fc Create .DS_Store
f 6bb4440 2019-07-08
f 41e11ce Assignments
f 43e4eff 2019-07-08
f aa63389 2019-07-09
f a2d2a0c Delete 0-96ppt.pdf
f c432e56 2019-07-09
f 2baac34 Delete 0-96ppt.pdf
f 4617c56 Update .DS_Store
f cb1ca78 2019-07-10
f 1291efe 2019-07-11
```
then save. If command line uses `vi`, type `:wq` to save and quit.
3. `git log`. Check if there is only one commit ahead of `upstream/master`.

It's better to tidy up commit before you resolve conflicts...
To resolve conflict, I would use command line. BTW, your branch name is meaningless. it should represent a issue or a characteristic that your PR relates to. For example, branch name can be `6_mission_2019_07_10`.
1. `git checkout Mengqi29`
2. `git fetch upstream`
3. `git rebase upstream/master`
4. It will give you what files have conflicts. Each line starts with `CONFLICT <PATH>`. Go to that file and resolve the conflict by accept/discard correct branch code from HEAD/Incoming. Then save files. For `.DS_Store`, just remove this file.
5. After resolving all the conflict, do `git add .`
6. `git rebase --continue`
7. `git log` to check if `(upstream/master)` is one commit older than your commit.
8. if it is, `git push -f`. If not, find the problem.

File: Assignments/2019-07-10/LC349_Efficient_Solution.java (hunk 1 above)
Comment (Mengqi29) [reply Assignments/2019-07-10/LC349_Efficient_Solution.java]: 好的！

**Pull request merged**